### PR TITLE
added controller , mapper, dto

### PR DESCRIPTION
**Pull request by dmytrocherepov**

### CODE REVIEW

**Review by VolodymyrSkvira — APPROVED** [review]
LGTM

File: pom.xml [hunk 1]
```diff
@@ -73,6 +80,30 @@
...
+						<path>
+							<groupId>org.projectlombok</groupId>
+							<artifactId>lombok-mapstruct-binding</artifactId>
+							<version>${lombok.mapstruct.binding.version}</version>
+						</path>
+						<path>
+							<groupId>org.mapstruct</groupId>
+							<artifactId>mapstruct-processor</artifactId>
+							<version>${mapstruct.version}</version>
+						</path>
```
Comment (liliia-ponomarenko) [inline pom.xml]: Let’s  fix spaces here;)

File: src/main/java/com/example/bookstoreapp/config/MapperConfig.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import org.mapstruct.InjectionStrategy;
+import org.mapstruct.NullValueCheckStrategy;
+
+@org.mapstruct.MapperConfig(
+        componentModel = "spring",
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        implementationPackage = "<PACKAGE_NAME>.impl"
+)
+public class MapperConfig {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/bookstoreapp/config/MapperConfig.java:12]: it is better to specify @MapperConfig directly on mapper class

File: src/main/java/com/example/bookstoreapp/controller/BookController.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/books")
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/bookstoreapp/controller/BookController.java]:
```suggestion
@RequestMapping("/books")
```
for now better not to specify it as api)

File: src/main/java/com/example/bookstoreapp/dto/BookDto.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.example.bookstoreapp.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class BookDto {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/bookstoreapp/dto/BookDto.java]: using records for dto is better practice. Let’s  use it for  each dto if it is possible

File: src/main/java/com/example/bookstoreapp/mapper/BookMapper.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import com.example.bookstoreapp.dto.CreateBookRequestDto;
+import com.example.bookstoreapp.model.Book;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+    BookDto toDto(Book book);
+
+    Book toModel(CreateBookRequestDto requestDto);
+
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/bookstoreapp/mapper/BookMapper.java]:
```suggestion
```

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  fix a few mistakes ;)

File: src/main/java/com/example/bookstoreapp/dto/BookDto.java (hunk 4 above)
Comment (dmytrocherepov) [reply src/main/java/com/example/bookstoreapp/dto/BookDto.java]: tried to use record on dto class , but maven throws error

File: pom.xml [hunk 6]
```diff
@@ -1,99 +1,129 @@
...
+                            <artifactId>mapstruct-processor</artifactId>
+                            <version>${mapstruct.version}</version>
+                        </path>
+                    </annotationProcessorPaths>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <version>3.1.1</version>
```
Comment (Elena-Bruyako) [inline pom.xml]: Update to 3.3.0 to use record in your code

File: src/main/java/com/example/bookstoreapp/mapper/BookMapper.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+import com.example.bookstoreapp.dto.BookDto;
+import com.example.bookstoreapp.dto.CreateBookRequestDto;
+import com.example.bookstoreapp.model.Book;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+    BookDto toDto(Book book);
+
+    Book toModel(CreateBookRequestDto requestDto);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstoreapp/mapper/BookMapper.java]: I would suggest toBook()

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bookstoreapp/dto/CreateBookRequestDto.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+        Long id,
+        String title,
+        String author,
+        String isbn,
+        BigDecimal price,
+        String description,
+        String coverImage
+) {
+}
+
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstoreapp/dto/CreateBookRequestDto.java:15]: remove this line

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**